### ISSUE TITLE
fix TestReplicaDataIterator

### ISSUE THREAD

**Issue by tschottdorf** [cockroachdb/cockroach#2851]
which sits in `./storage/replica_data_iter_test.go`. It was written at a time in which a `Replica` would just sit still after having been created. Nowadays, such a `Replica` will elect itself as Raft leader though, which writes state.

For fixing the test, we should probably relax the rigidity of the check, in particular for Raft-related keys (or whatever other moving parts there might be).

**Comment by lizhongz** [comment]
Happy to fix this :-)

**Comment by tschottdorf** [comment]
Great! Since the situation here isn't as clear, maybe post your plan before attempting a fix - that way we can give you feedback early.

**Comment by lizhongz** [comment]
@tschottdorf sure. That sounds good.

**Comment by lizhongz** [comment]
@tschottdorf Here is my fix plan.
1 Switch the order of two keys.RaftLogKey() in createRangeData() in replica_data_iter_test.go, as follows.
```go
{keys.RaftLogKey(r.Desc().RangeID, 1), ts0},
{keys.RaftLogKey(r.Desc().RangeID, 2), ts0},
```
Because keys are sorted in a range.

2 After destroying a replica, all keys were removed but a tombstone key was added. So the replica is not empty. Therefore, we can compare the key to a tombstone key here to fix this.
```go
    // Destroy range and verify that its data has been completely cleared.
    if err := tc.rng.Destroy(); err != nil {
        t.Fatal(err)
    }
    iter = newReplicaDataIterator(tc.rng.Desc(), tc.rng.rm.Engine())
    defer iter.Close()
    if iter.Valid() {
        k1, ts1, _, err := engine.MVCCDecodeKey(iter.Key())
        if err != nil {
            t.Fatal(err)
        }
        if bytes.Compare(k1, keys.RaftTombstoneKey(tc.rng.Desc().RangeID)) != 0 {
            t.Errorf("expected empty iteration; got first key %q, ts %q", k1, ts1)
        }
    }
```

**Comment by tschottdorf** [comment]
Sounds good. Sorry I didn't get back to your earlier - I'll be commenting
on your PR tomorrow.

**Comment by tamird** [comment]
Fixed in #2959.
